test/issues/reparce_tag_file_placeholder.cpp

```diff
@@ -0,0 +1,139 @@
+//  Boost operations_test.cpp  ---------------------------------------------------------//
+
+//  Copyright Roman Savchenko 2020
+
+//  Distributed under the Boost Software License, Version 1.0.
+//  See http://www.boost.org/LICENSE_1_0.txt
+
+//  Library home page: http://www.boost.org/libs/filesystem
+
+#include <boost/scope_exit.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/core/lightweight_test.hpp>
+
+#include <iostream>
+
+# include <windows.h>
+# include <winnt.h>
+
+#ifdef _MSC_VER
+#  pragma comment(lib, "Advapi32.lib")
+#endif
+
+// Test correct boost::filesystem::status when reparse point ReparseTag set to IO_REPARSE_TAG_FILE_PLACEHOLDER
+// https://docs.microsoft.com/en-us/windows/compatibility/placeholder-files?redirectedfrom=MSDN
+
+#ifndef IO_REPARSE_TAG_FILE_PLACEHOLDER
+#  define IO_REPARSE_TAG_FILE_PLACEHOLDER (0x80000015L)
+#endif
+
+typedef struct _REPARSE_DATA_BUFFER {
+  ULONG  ReparseTag;
+  USHORT  ReparseDataLength;
+  USHORT  Reserved;
+  union {
+    struct {
+      USHORT  SubstituteNameOffset;
+      USHORT  SubstituteNameLength;
+      USHORT  PrintNameOffset;
+      USHORT  PrintNameLength;
+      ULONG  Flags;
+      WCHAR  PathBuffer[1];
+    } SymbolicLinkReparseBuffer;
+    struct {
+      USHORT  SubstituteNameOffset;
+      USHORT  SubstituteNameLength;
+      USHORT  PrintNameOffset;
+      USHORT  PrintNameLength;
+      WCHAR  PathBuffer[1];
+    } MountPointReparseBuffer;
+    struct {
+      UCHAR  DataBuffer[1];
+    } GenericReparseBuffer;
+  };
+} REPARSE_DATA_BUFFER, *PREPARSE_DATA_BUFFER;
+
+#define REPARSE_DATA_BUFFER_HEADER_SIZE \
+  FIELD_OFFSET(REPARSE_DATA_BUFFER, GenericReparseBuffer)
+
+bool obtain_restore_privilege()
+{
+  HANDLE hToken;
+  if (!OpenProcessToken(GetCurrentProcess(), TOKEN_ADJUST_PRIVILEGES, &hToken))
+  {
+    std::cout << "OpenProcessToken() failed with: " << GetLastError() << std::endl;
+    return false;
+  }
+
+  BOOST_SCOPE_EXIT(&hToken)
```

Comment:
Better not use the ScopeExit library. Just write a simple scope guard class.

Reply:
done